source/tilecameragenerator.ts

```diff
@@ -0,0 +1,491 @@
+        this.genHilbertIndices(0, 0, tableSizeNextPowerOfTwo, 0, 0, tableSizeNextPowerOfTwo, recursionDepth);
+    }
+
+    /**
+     * Recursively fills the iterationAlgorithmIndices with the HilbertCurve.
+     * Do not use this method. Use fillIterationIndicesWithHilbert() instead.
+     */
+    protected genHilbertIndices(x: number, y: number,
+        xi: number, xj: number, yi: number, yj: number, depth: number): void {
+        if (depth <= 0) {
+            x = x + (xi + yi - 1) / 2;
+            y = y + (xj + yj - 1) / 2;
+            if (x < this.numberOfXTiles() && y < this.numberOfYTiles()) {
+                this._iterationAlgorithmIndices.push([x, y]);
+            }
+        } else {
+            this.genHilbertIndices(x, y, yi / 2, yj / 2, xi / 2, xj / 2, depth - 1);
+            this.genHilbertIndices(x + xi / 2, y + xj / 2, xi / 2, xj / 2, yi / 2, yj / 2, depth - 1);
+            this.genHilbertIndices(x + xi / 2 + yi / 2, y + xj / 2 + yj / 2, xi / 2, xj / 2, yi / 2, yj / 2, depth - 1);
+            this.genHilbertIndices(x + xi / 2 + yi, y + xj / 2 + yj, -yi / 2, -yj / 2, -xi / 2, -xj / 2, depth - 1);
+        }
+    }
+
+    /**
+     * Returns the padded tileSize.
+     * @returns - The padded tileSize.
+     */
+    protected getPaddedTileSize(): [number, number] {
+        return [this.padding[1] + this.padding[3] + this.tileSize[0],
+        this.padding[0] + this.padding[2] + this.tileSize[1]];
+    }
+
+    /**
+     * Used for Iterator behavior. It sets the tile to 0 (first tile) if the value of tile is negative
+     * and therefore initiates the iteration.
+     * Otherwise it increments the tile and calls update to directly change the camera.
+     * If the tile index would get out of range it returns false to indicate, that all tiles were rendered.
+     * @returns - If the camera has been set to a next tile.
+     */
+    public nextTile(): boolean {
+        if (this.tile >= this.numberOfTiles() - 1) {
+            return false;
+        }
+        if (this.tile < 0) {
+            this.tile = -1;
+        }
+        ++this.tile;
+        this.update();
+        return true;
+    }
+
+    /**
+     * Returns if tiles still need to be rendered.
+     */
+    public hasNextTile(): boolean {
+        if (this.tile < this.numberOfTiles() - 1 && this.tile >= 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Resets the tile index to prepare the generator for the next rendering.
+     * Should be called after iteration with nextTile().
+     */
+    public resetTileRendering(): void {
+        this.tile = -1;
+        this._currentOffset[0] = 0;
+        this._currentOffset[1] = 0;
+    }
+
+    /**
+     * Reassigns all values from the source camera to the tile camera.
+     * Should be called when the source camera is altered.
+     */
+    public updateCameraProperties(): void {
+        this.sourceCamera.copyAllValues(this.camera);
```

Comment:
This indicated that `sourceCamera` and `camera` should always hold similiar or even same values. Why do we still use a copy?

Reply:
Since we do not want to manipulate the camera provided by the user (```sourceCamera```). We change the ```postViewProjection``` on our camera (```camera```) to render the current tile, which would alter the users camera.
And yes it is true that the the only difference between these camera is the ```postViewProjection```.